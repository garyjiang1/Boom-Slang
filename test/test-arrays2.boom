int[] arr = [3+2, exp(5,2), 42]

print_arr3(arr)

# this function calculates b^p
def exp(int b, int p) returns int:
	if p == 0:
		return b
	int start = 1
	int i = 0
	loop i += 1 while i < p:
		start *= b
	return start

# this function prints out an array of size 3
def print_arr3(int[] a) returns void:
	int i = 0
	loop i += 1 while i < 3:
		println(arr[i])

# make everyting in arr default 
arr = default int[3]
println("")
print_arr3(arr)
