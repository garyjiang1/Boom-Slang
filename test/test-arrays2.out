5
25
42

0
0
0
